code/game/objects/items/weapons/ecigs.dm

```diff
@@ -0,0 +1,320 @@
+/obj/item/clothing/mask/smokable/ecig
+	name = "electronic cigarette"
+	desc = "Device with modern approach to smoking."
+	icon = 'icons/obj/ecig.dmi'
```

Comment:
this should be a contained sprite, you can look into how to make species specific sprites by looking at how hardsuits are done